src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseSimulationService.java

```diff
@@ -0,0 +1,162 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.domain.enumeration.BuildPlanType.SOLUTION;
+import static de.tum.in.www1.artemis.domain.enumeration.BuildPlanType.TEMPLATE;
+
+import java.util.Optional;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.enumeration.AssessmentType;
+import de.tum.in.www1.artemis.domain.enumeration.RepositoryType;
+import de.tum.in.www1.artemis.domain.enumeration.SubmissionType;
+import de.tum.in.www1.artemis.domain.participation.SolutionProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.participation.TemplateProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.scheduled.ProgrammingExerciseScheduleService;
+import de.tum.in.www1.artemis.service.util.VCSSimulationUtils;
+
+@Profile("dev")
+@Service
+public class ProgrammingExerciseSimulationService {
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingExerciseScheduleService programmingExerciseScheduleService;
+
+    private final GroupNotificationService groupNotificationService;
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final TemplateProgrammingExerciseParticipationRepository templateProgrammingExerciseParticipationRepository;
+
+    private final SolutionProgrammingExerciseParticipationRepository solutionProgrammingExerciseParticipationRepository;
+
+    private final ProgrammingSubmissionRepository programmingSubmissionRepository;
+
+    private final ResultRepository resultRepository;
+
+    public final String domain = "nolocalsetup:7990/scm/";
```

Comment:
I would prefer something like https://artemislocal.ase.in.tum.de

Reply:
or something like artemislocalhost:7990/scm/

Reply:
i changed the URL. The new URL is artemislocalhost:7990/scm/